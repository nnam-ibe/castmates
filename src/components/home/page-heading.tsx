export const PageHeading = () => {
  return (
    <div className="text-center space-y-2">
      <h1 className="text-3xl font-bold">Actor Collaboration Explorer</h1>
      <p className="text-neutral-600">
        Compare filmographies and discover shared movies
      </p>
    </div>
  );
};
