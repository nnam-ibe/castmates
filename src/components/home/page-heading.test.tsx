import { describe, test, expect, beforeEach } from "vitest";
import { render, screen } from "@testing-library/react";
import { PageHeading } from "./page-heading";

describe("PageHeading", () => {
  beforeEach(() => {
    render(<PageHeading />);
  });

  test("renders the main heading with correct text and styling", () => {
    const heading = screen.getByRole("heading", {
      level: 1,
      name: /actor collaboration explorer/i,
    });

    expect(heading.className).toContain("text-3xl");
    expect(heading.className).toContain("font-bold");
  });

  test("renders the subtitle with correct text and styling", () => {
    const subtitle = screen.getByText(
      /compare filmographies and discover shared movies/i
    );

    expect(subtitle.className).toContain("text-neutral-600");
  });

  test("container has correct layout classes", () => {
    const container = screen.getByRole("heading", {
      level: 1,
      name: /actor collaboration explorer/i,
    }).parentElement;

    expect(container?.className).toContain("text-center");
    expect(container?.className).toContain("space-y-2");
  });

  test("matches snapshot", () => {
    const { container } = render(<PageHeading />);
    expect(container).toMatchSnapshot();
  });
});
