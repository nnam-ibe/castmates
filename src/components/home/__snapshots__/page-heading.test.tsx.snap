// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageHeading > matches snapshot 1`] = `
<div>
  <div
    class="text-center space-y-2"
  >
    <h1
      class="text-3xl font-bold"
    >
      Actor Collaboration Explorer
    </h1>
    <p
      class="text-neutral-600"
    >
      Compare filmographies and discover shared movies
    </p>
  </div>
</div>
`;
